Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/AzureCliAzureManager.java

```diff
@@ -0,0 +1,236 @@
+            try {
+                getInstance().signIn();
+            } catch (AzureExecutionException e) {
+                // Catch the exception when restore
+            } finally {
+                return authMethodDetails;
+            }
+        }
+    }
+
+    public static AzureCliAzureManager getInstance() {
+        return LazyLoader.INSTANCE;
+    }
+
+    @Override
+    public Azure getAzure(String sid) throws IOException {
+        if (sidToAzureMap.containsKey(sid)) {
+            return sidToAzureMap.get(sid);
+        }
+        final Azure azure = auth().withSubscription(sid);
+        // TODO: remove this call after Azure SDK properly implements handling of unregistered provider namespaces
+        AzureRegisterProviderNamespaces.registerAzureNamespaces(azure);
+        sidToAzureMap.put(sid, azure);
+        return azure;
+    }
+
+    @Override
+    public AppPlatformManager getAzureSpringCloudClient(String sid) {
+        return sidToAzureSpringCloudManagerMap.computeIfAbsent(sid, s ->
+                buildAzureManager(AppPlatformManager.configure()).authenticate(azureCliCredentials, s));
+    }
+
+    @Override
+    public InsightsManager getInsightsManager(String sid) {
+        return sidToInsightsManagerMap.computeIfAbsent(sid, s ->
+                buildAzureManager(InsightsManager.configure()).authenticate(azureCliCredentials, s));
+    }
+
+    @Override
+    public List<Subscription> getSubscriptions() {
+        return authenticated.subscriptions().list();
+    }
+
+    @Override
+    public List<Pair<Subscription, Tenant>> getSubscriptionsWithTenant() {
+        final Tenant subscriptionTenant = authenticated.tenants().list().stream()
+                .filter(tenant -> StringUtils.equals(tenant.tenantId(), authenticated.tenantId()))
+                .findFirst().orElse(null);
+        final List<Pair<Subscription, Tenant>> result = new ArrayList<>();
+        for (Subscription subscription : getSubscriptions()) {
+            result.add(new Pair<>(subscription, subscriptionTenant));
+        }
+        return result;
+    }
+
+    @Override
+    public Settings getSettings() {
+        return settings;
+    }
+
+    @Override
+    public SubscriptionManager getSubscriptionManager() {
+        return subscriptionManager;
+    }
+
+    @Override
+    public void drop() throws IOException {
+        authenticated = null;
+        azureCliCredentials = null;
+        subscriptionManager.cleanSubscriptions();
+    }
+
+    @Override
+    public KeyVaultClient getKeyVaultClient(String tid) {
+        final ServiceClientCredentials credentials = new KeyVaultCredentials() {
+            @Override
+            public String doAuthenticate(String authorization, String resource, String scope) {
+                try {
+                    return azureCliCredentials.getToken(resource);
+                } catch (Exception ex) {
+                    throw new RuntimeException(ex);
```

Comment:
Where to invoke `doAuthenticate()`? Looks like the exception needs to be `unbox` there.

Reply:
Will return null if not authenticated, `doAuthenticate` should be invoked by KeyValult SDK itself.